File: BOJ/python/1stAgain/회의실배정Greedy.py
import sys
from collections import deque
sys.setrecursionlimit(10000)

if __name__ == "__main__" :
    
    n = int(input())
    meeting = []
    
    for i in range(n):
        s, e = map(int,input().split())
        meeting.append((s,e))

    # end 시간 기준 정렬
    meeting.sort(key = lambda x : (x[1], x[0]))

    start = 0
    end   = 0
    res   = 0
    
    for st, ed in meeting :
        if st >= end :
            res += 1
            end   = ed
    print(res)
            
            
            
                
            
    

File: Algorithm/BOJ/js/회의실배정Greedy.js
let fs = require('fs');
// var input = fs.readFileSync('/dev/stdin').toString().trim().split('\r\n');
let input = fs.readFileSync('js/test.txt').toString().trim().split('\r\n');
let NumOfMeeting = parseInt(input[0])
let Meetings = []
input.shift()


// 회의 정보 입력받기
for(let i = 0 ; i < NumOfMeeting ; i++){
    input[i] = input[i].split(' ')
    Meetings.push([parseInt(input[i][0]), parseInt(input[i][1])])
}

Meetings.sort((a,b) => {
    return a[1] - b[1]
})

let start = 0 
let end   = 0
let FinalNum = 0 

Meetings.forEach((meeting) => {
    if( meeting[0] >= end){
        end = meeting[1]
        FinalNum += 1
    }
})

console.log(FinalNum)